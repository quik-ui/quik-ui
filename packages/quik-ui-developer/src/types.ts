import { ReactNode } from "react";

export type Components = Component[];

type Component = {
  component: ReactNode | (() => ReactNode);
  spec: ComponentSpec;
};

type ComponentSpec = {
  name: string;
  overview?: string;
  variants?: ComponentVariants;
  props?: ComponentProps;
};

type ComponentOverview = {
  version: string;
  description: string;
  props: ComponentOverviewProps;
};

type ComponentOverviewProps = {
  [key: string]: {
    name: string;
    description: string;
    type: string;
    required: boolean;
    defaultValue: string;
    control: string;
  };
};

type ComponentProps = {
  [key: string]: any[];
};

type ComponentVariants = {
  [key: string]: object;
};
