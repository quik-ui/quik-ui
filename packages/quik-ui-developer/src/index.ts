function Hello() {
  console.log("Hello, Quik UI!");
}

export { Hello };
