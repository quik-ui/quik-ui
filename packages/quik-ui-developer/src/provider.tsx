import React, { Dispatch, ReactNode, useEffect, useReducer } from "react";
import { DevSettings } from "react-native";

const DeveloperContext = React.createContext<ReducerState | undefined>(
  undefined,
);

const DeveloperUpdateContext = React.createContext<
  Dispatch<ReducerActions> | undefined
>(undefined);

type DeveloperProviderProps = {
  children: ReactNode;
};

type ReducerState = {
  menuItemAdded: boolean;
  visible: boolean;
};

type ReducerActions =
  | {
      type: "TOGGLE_VISIBILITY";
    }
  | { type: "ADD_MENU_ITEM" };

const initialState: ReducerState = {
  menuItemAdded: false,
  visible: false,
};

const reducer = (state: ReducerState, action: ReducerActions): ReducerState => {
  switch (action.type) {
    case "TOGGLE_VISIBILITY":
      return { ...state, visible: !state.visible };
    case "ADD_MENU_ITEM":
      return { ...state, menuItemAdded: true };
    default:
      return state;
  }
};

const DeveloperProvider = ({ children }: DeveloperProviderProps) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  useEffect(() => {
    return () => {
      if (state.menuItemAdded) return;
      DevSettings.addMenuItem("Toggle Developer", () => {
        dispatch({ type: "TOGGLE_VISIBILITY" });
      });
      dispatch({ type: "ADD_MENU_ITEM" });
    };
  }, []);

  return (
    <DeveloperContext.Provider value={state}>
      <DeveloperUpdateContext.Provider value={dispatch}>
        {children}
      </DeveloperUpdateContext.Provider>
    </DeveloperContext.Provider>
  );
};

function useDeveloper() {
  const context = React.useContext(DeveloperContext);
  if (context === undefined) {
    throw new Error("useDeveloper must be used within a DeveloperProvider");
  }
  return context;
}

function useDeveloperUpdate() {
  const context = React.useContext(DeveloperUpdateContext);
  if (context === undefined) {
    throw new Error(
      "useDeveloperUpdate must be used within a DeveloperProvider",
    );
  }
  return context;
}

function useToggleDeveloper() {
  const dispatch = useDeveloperUpdate();
  return () => dispatch({ type: "TOGGLE_VISIBILITY" });
}
