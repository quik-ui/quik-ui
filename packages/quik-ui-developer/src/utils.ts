import { lazy } from "react";

function loadComponents(...args: string[]) {
  return args.map((name) => {
    return lazy(() => import(name));
  });
}
