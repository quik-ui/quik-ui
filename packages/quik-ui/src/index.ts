export * from "./components/Container/Container";

export * from "./core/theme/context";
export * from "./core/theme/provider";
export * from "./core/theme/utils";
export * from "./core/theme/theme";

export type * from "./types/colors";
export type * from "./types/component";
export type * from "./types/fonts";
export type * from "./types/spacing";
export type * from "./types/theme";
