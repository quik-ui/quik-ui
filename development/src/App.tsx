import React from 'react';
import {ThemeProvider, useCreateTheme} from 'quik-ui';

function App() {
  const {theme} = useCreateTheme({mode: 'light'});

  return <ThemeProvider theme={theme}></ThemeProvider>;
}
export default App;
