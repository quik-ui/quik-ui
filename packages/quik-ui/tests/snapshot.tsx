import React from "react";
import renderer from "react-test-renderer";

import { Container } from "../src";

test("renders correctly", () => {
  const tree = renderer.create(<Container />).toJSON();
  expect(tree).toMatchSnapshot();
});
