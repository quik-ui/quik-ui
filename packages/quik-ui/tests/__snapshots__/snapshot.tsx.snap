// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "blue",
      "flex": 1,
    }
  }
/>
`;
